### PR TITLE
Rewrite test workflow with agent and allure dirs

### PR DESCRIPTION
**Pull request by gigimon**
Rewrite workflow:

1. Rename to more clearly name
2. Remove unused "second" allure report
3. Use hosted runner and cleanup
4. Use allure directories based on stand and branch
5. Use select box for run options
6. Run openzeppelin tests in N threads once (not for each file)

https://docs.neon-labs.org/neon-compatibility/feature/experiments-with-jobs/244/

Test was longer because in 4 thread, change to 8 by default.

### CODE REVIEW

File: .github/workflows/consecutive.yml [hunk 1]
```diff
@@ -1,33 +1,74 @@
...
-        description: "Test scenario tags"
-        required: false
       network:
-        description: "Network (devnet, testnet, testnet-aws, teststand, teststand2, neonswap.live, aws, ropsten, rinkeby)"
+        type: choice
+        default: teststand2
         required: true
-        default: "teststand2"
+        description: "Network (devnet, testnet, teststand, teststand2, neonswap.live, aws, ropsten, rinkeby)"
+        options:
```
Comment (apetrovskiy) [inline .github/workflows/consecutive.yml:13]: image

File: .github/workflows/consecutive.yml [hunk 2]
```diff
@@ -1,33 +1,74 @@
-name: "GNU Parallel: Python 3.8, Node.js 16, allure"
+name: "Run compatibility tests (python, node.js, openzeppelin)"
```
Comment (apetrovskiy) [inline .github/workflows/consecutive.yml:1]: just a note: looks like the multi-job workflow is not about running compatibility tests :)

File: .github/workflows/consecutive.yml [hunk 3]
```diff
@@ -267,20 +212,33 @@ jobs:
...
         uses: neonlabsorg/allure-report-action@master
         if: always()
         with:
           allure_results: allure-results
           gh_pages: gh-pages
+          subfolder: ${{ env.ALLURE_SUBDIR }}
           allure_report: allure-report
           allure_history: allure-history
-          keep_reports: 30
+          keep_reports: 50
```
Comment (apetrovskiy) [inline .github/workflows/consecutive.yml:236]: why do you need 50? isn't it working too slowly?

File: hardhat.config.js [hunk 4]
```diff
@@ -59,7 +59,7 @@ const withOptimizations =
 
 const ACCOUNTS_NUMBER = parseInt(process.env.USERS_NUMBER);
 
-const web3 = new Web3(new Web3.providers.HttpProvider(process.env.PROXY_URL));
+const web3 = new Web3(new Web3.providers.HttpProvider(process.env.PROXY_URL, 3000000));
```
Comment (apetrovskiy) [inline hardhat.config.js:62]: why do we need it? have we ever fallen here?
if you want to add one more timeout expiration value, wouldn't it be better to create an entry in .env files or move this setting into a constant class?

File: .github/workflows/consecutive.yml [hunk 5]
```diff
@@ -63,154 +98,60 @@ jobs:
...
         if: always()
       - run: npm run build --if-present
+      - name: "Run openzeppelin in parallel"
+        run: |
+          cd openzeppelin-contracts
+          cp ../run_parallel.py ./
+          cp ../rrrun.sh ./
+          chmod +x rrrun.sh
+          npx hardhat compile
+          python3 run_parallel.py
```
Comment (apetrovskiy) [inline .github/workflows/consecutive.yml:131]: why do we need this instead of GNU parallel?

File: .github/workflows/consecutive.yml (hunk 2 above)
Comment (gigimon) [reply .github/workflows/consecutive.yml:1]: I think "Run <smth>" is more clearly than "GNU Parallel"

File: .github/workflows/consecutive.yml (hunk 3 above)
Comment (gigimon) [reply .github/workflows/consecutive.yml:236]: No, this momen doesn't affect speed. I increase this for save same logs in branches.

File: hardhat.config.js (hunk 4 above)
Comment (gigimon) [reply hardhat.config.js:62]: We can remove this. I set this when found some timeout which we can increase

File: .github/workflows/consecutive.yml (hunk 5 above)
Comment (gigimon) [reply .github/workflows/consecutive.yml:131]: command in python script - parallel. This script uses for run only N copies tests and split all test files for N processes. Not every file in new node.js

**Comment by gigimon** [comment]
@apetrovskiy what result? can merge? :)

File: .github/workflows/consecutive.yml [hunk 6]
```diff
@@ -63,54 +98,60 @@ jobs:
...
-          cat $GITHUB_ENV
-        shell: bash
-      # macos-11
-      # - uses: ./.github/actions/allure-prep
       - uses: ./.github/actions/openzeppelin-preparation
-      - uses: ./.github/actions/switch-network
-        if: always()
+      - name: Switch network
+        run: |
+          cp "./.env.$NETWORK" "./openzeppelin-contracts/.env"
```
Comment (apetrovskiy) [inline .github/workflows/consecutive.yml]: why do we need this?

File: .github/workflows/consecutive.yml [hunk 7]
```diff
@@ -63,54 +98,55 @@ jobs:
         with:
           folder-name: report/allure-results
 
-  node-contracts:
-    runs-on: neon-hosted
-    timeout-minutes: 361
-    if: always()
+  openzeppelin:
+    timeout-minutes: 360
```
Comment (apetrovskiy) [inline .github/workflows/consecutive.yml:102]: it's the default value, the same as we omit the timeout-minutes parameter at all.
I noticed a couple of time that GHA job got frozen and, even I don't know how 'software from Ivanovo' works internally, just tried to set a limit a bit bigger than possible for a job

File: .github/workflows/consecutive.yml [hunk 8]
```diff
@@ -237,3 +274,12 @@ jobs:
...
           networkName: ${{ env.NETWORK_NAME }}
+
+  cleaner:
+    runs-on: self-hosted
+    needs: notification
+    if: always()
+    steps:
+      - name: Cleanup runner
+        run: | 
+          echo "Cleaning up previous run" rm -rf "${{ github.workspace }}"
```
Comment (apetrovskiy) [inline .github/workflows/consecutive.yml]: Wouldn't it better to split this into two lines of code?

File: .github/workflows/consecutive.yml [hunk 9]
```diff
@@ -147,30 +183,18 @@ jobs:
         with:
           folder-name: report/allure-results
 
-  clean-up:
```
Comment (apetrovskiy) [inline .github/workflows/consecutive.yml:150]: btw, this cleaner has already been tested

File: .github/workflows/consecutive.yml (hunk 7 above)
Comment (gigimon) [reply .github/workflows/consecutive.yml:102]: In self-hosted runner job can be run more 6 hours (up to 24).

**Review by apetrovskiy — APPROVED** [review]

**Pull request merged**